README.md

```diff
@@ -514,6 +514,86 @@ $ grpcurl -plaintext 127.0.0.1:9093 spacemesh.v1.DebugService.NetworkInfo
+```
+
+In this case, an empty line and `-- some change` was added to `schema.sql` by hand. The pretty-printed diff looks like this:
+```
+  (
+  	"""
+  	... // 81 identical lines
+  	    PRIMARY KEY (kind, epoch)
+  	) WITHOUT ROWID;
+-
+- 	-- some change
+  	"""
+  )
+```
+
+The possible reasons for schema drift can be the following:
+* running an unreleased version of go-spacemesh using your data folder. The unreleased version may contain migrations that may be changed before the release happens
+* manual changes in the database
+* external SQLite tooling used on the database that adds some tables, indices etc.
+
+In the latter case, it is possible to make go-spacemesh ignore certain objects (tables and indices) when checking for schema drift. For this, you can use `main.db-schema-ignore-rx` setting to set a regular expression that is used to ignore tables and indices in the database during schema drift checks. The setting defaults to `_litestream` to help with certain tooling.
+
+The schema changes in go-spacemesh code should be always done by means of adding migrations. After that, the schema tests in `sql/localsql` and `sql/statesql` will start failing. When the tests fail, they display the difference between the schema stored in `schema.sql` and the schema that is loaded from the database after running all the migrations.
+If the schema changes shown in the diff are expected, the schema file needs to be updated.
+
+```console
+$ # run the tests
+$ eval $(make print-test-env) go test ./sql/localsql ./sql/statesql
+...
+=== RUN   TestSchema/schema/force_migrations
+    test.go:106: updated schema written to schema/schema.sql.updated
+    test.go:108: 
+        	Error Trace:	/Users/ivan4th/work/spacemesh/go-spacemesh/sql/test/test.go:108
+        	Error:      	Should be empty, but was   (
+        	            	  	"""
+        	            	  	... // 81 identical lines
+        	            	  	    PRIMARY KEY (kind, epoch)
+        	            	  	) WITHOUT ROWID;
+        	            	- 	-- some change
+        	            	  	"""
+        	            	  )
+        	Test:       	TestSchema/schema/force_migrations
+        	Messages:   	schema diff
+FAIL
+FAIL    github.com/spacemeshos/go-spacemesh/sql/localsql        0.163s
+ok      github.com/spacemeshos/go-spacemesh/sql/statesql        0.286s
+FAIL
+$ git status
+...
+Untracked files:
+  (use "git add <file>..." to include in what will be committed)
+        sql/localsql/schema/schema.sql.updated
+
+$ # update the schema file
+$ mv sql/localsql/schema/schema.sql{.updated,}
+
+$ # rerun the tests
+$ eval $(make print-test-env) go test -count=1 ./sql/localsql ./sql/statesql
+ok      github.com/spacemeshos/go-spacemesh/sql/localsql        0.166s
+ok      github.com/spacemeshos/go-spacemesh/sql/statesql        0.293s
+```
```

Comment:
No user will clone our repository and run tests if their node stops working due to a schema drift - I think this should be removed from the `README.md`. Also you are kinda leaking information about your system with putting or workspace path into the README 😉

Reply:
Updated schema handling docs and moved them to `CONTRIBUTING.md`